Zend/zend_atomic.h

```diff
@@ -23,8 +23,8 @@
 	((__GNUC__ == (x) && __GNUC_MINOR__ >= (y)) || (__GNUC__ > (x)))
 
 /* Builtins are used to avoid library linkage */
-#if __has_feature(c_atomic)
-#define	HAVE_C11_ATOMICS 1
+#if __has_feature(c_atomic) && defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201710L
+#define	HAVE_C17_ATOMICS 1
```

Comment:
I've updated the linked issue with the compiler and OS information.

What makes you say this wouldn't be suitable for PHP 8.2? I have already applied this patch to MacPorts PHP 8.2 and 8.3 and have not received any reports of problems, though of course that doesn't guarantee there aren't any problems.

Reply:
We cannot just take away `HAVE_C11_ATOMICS` existing in a patch release.

Reply:
If your objection is about changing the name if the macro from `HAVE_C11_ATOMICS` to `HAVE_C17_ATOMICS`, then a solution could be not to change the name.

Reply:
I renamed it in order to have the macro name reflect the change in its definition. I did not consider that it might be part of public API that others would use. If it is public API, then another option might be to leave `HAVE_C11_ATOMICS` defined as it was and just not use it in PHP itself, addition to creating `HAVE_C17_ATOMICS` and using it as in this PR.

Of course, I don't care how it's fixed, and I'm not very familiar with the code of the PHP project so I don't know what fix is best. All I care about is that it is fixed somehow.